core/src/main/java/org/apache/calcite/sql/SqlUtil.java

```diff
@@ -1189,4 +1189,19 @@ private void visitChild(SqlNode node) {
       return check(type);
     }
   }
+
+  /**
+   * Whether the selectItem is a table function node in the select.
+   * eg. "select table_func(1) as (f0,f1)"
+   *
+   * @param selectItem select item
+   * @return true if this selectItem is a table function call
+   */
+  public static boolean isTableFunctionInSelect(SqlNode selectItem) {
+    if (selectItem.getKind() == SqlKind.AS) {
+      SqlBasicCall call = (SqlBasicCall) selectItem;
```

Comment:
At least, we should check the function is a `SqlTableFunction`.

Reply:
If the second of the `AS` is a `SqlNodeList`, it implies the call is a `UDTF`. Otherwise, it will throw exception in the next step.

Reply:
A function's logic should be either a closure，or it should have a pre-check for its arguments.

Reply:
I changed the function name to make it clearer.